File: 1603. Design Parking System/C#/Design Parking System/ParkingSystem.cs
namespace Design_Parking_System;

public class ParkingSystem
{
    public ParkingSystem(int big, int medium, int small)
    {
        BigSlots = big;
        MediumSlots = medium;
        SmallSlots = small;
    }

    public int BigSlots { get; set; }
    public int MediumSlots { get; set; }
    public int SmallSlots { get; set; }

    public bool AddCar(int carType)
    {
        switch (carType)
        {
            case 1:
                BigSlots--;
                return BigSlots >= 0;
            case 2:
                MediumSlots--;
                return MediumSlots >= 0;
            case 3:
                SmallSlots--;
                return SmallSlots >= 0;
            default: return false;
        }
    }
}

File: leet-code/1603. Design Parking System/Python/parking_system.py
class ParkingSystem:
    small_slots = 0
    medium_slots = 0
    big_slots = 0

    def __init__(self, big_slots, medium_slots, small_slots):
        self.small_slots = small_slots
        self.medium_slots = medium_slots
        self.big_slots = big_slots

    def add_car(self, car_type):
        match car_type:
            case 1: 
                self.big_slots -= 1
                return self.big_slots >= 0
            case 2:
                self.medium_slots -= 1
                return self.medium_slots >= 0
            case 3:
                self.small_slots -= 1
                return self.small_slots >= 0